fs/zbd_zenfs.cc

```diff
@@ -711,6 +717,55 @@ IOStatus ZonedBlockDevice::AllocateEmptyZone(Zone **zone_out) {
   return IOStatus::OK();
 }
 
+int ZonedBlockDevice::DirectRead(char *buf, uint64_t offset, uint32_t n) {
+  int ret = 0;
+  int f = GetReadDirectFD();
+
+  while (ret < n) {
+    int r = pread(f, buf, n, offset);
+    if (r == -1) {
+      return -1;
+    }
+    if (r == 0) {
+      break;
+    }
+    ret += r;
+  }
+
+  return ret;
+}
+
+IOStatus ZonedBlockDevice::ReleaseMigrateZone(Zone *zone) {
+  {
+    std::unique_lock<std::mutex> lock_(migrate_zone_mtx_);
+    migrating_ = false;
+    if (zone != nullptr) {
+      zone->CheckRelease();
+      Info(logger_, "ReleaseMigrateZone: %lu", zone->start_);
+    }
+  }
+  migrate_resource_.notify_one();
```

Comment:
Before calling `notify_one`, we should let the unique_lock run out of scope. See https://en.cppreference.com/w/cpp/thread/condition_variable/notify_one

Reply:
Hmm... Here the unique_lock is already run out of scope, I think.

Reply:
You are right! I don't know why I did not see that :D